src/CreditCard.cs

```diff
@@ -5,79 +5,115 @@
+            set
+            {
+                ExpMonth = value.Month;
+                ExpYear = value.Year;
+            }
+        }
 
-		void IObjectValidation.AddParametersToRequest(RestRequest request)
-		{
-			request.AddParameter("card[number]", Number);
-			request.AddParameter("card[exp_month]", ExpMonth);
-			request.AddParameter("card[exp_year]", ExpYear);
-			if (Cvc.HasValue()) request.AddParameter("card[cvc]", Cvc);
-			if (Name.HasValue()) request.AddParameter("card[name]", Name);
-			if (AddressLine1.HasValue()) request.AddParameter("card[address_line1]", AddressLine1);
-			if (AddressLine2.HasValue()) request.AddParameter("card[address_line2]", AddressLine2);
-			if (AddressCity.HasValue()) request.AddParameter("card[address_city]", AddressCity);
-			if (AddressState.HasValue()) request.AddParameter("card[address_state]", AddressState);
-			if (AddressZip.HasValue()) request.AddParameter("card[address_zip]", AddressZip);
-			if (AddressCountry.HasValue()) request.AddParameter("card[address_country]", AddressCountry);
-		}
-	}
+        /// <summary>
+        /// Performs basic validation to make sure required values are present.
+        /// </summary>
+        void IObjectValidation.Validate()
+        {
+            Require.Argument("number", Number);
+            Require.Argument("exp_month", ExpMonth);
+            Require.Argument("exp_year", ExpYear);
+
+            Validate.IsBetween(ExpMonth, 1, 12);
+            Validate.IsBetween(ExpYear, DateTime.Now.Year, int.MaxValue);
+        }
+
+        void IObjectValidation.AddParametersToRequest(RestRequest request)
+        {
+            if (Token.HasValue())
+            {
+                request.AddParameter("source", Token);
+            }
+            else
+            {
+                request.AddParameter("source[object]", "card");
+                request.AddParameter("source[number]", Number);
+                request.AddParameter("source[exp_month]", ExpMonth);
+                request.AddParameter("source[exp_year]", ExpYear);
+                if (Cvc.HasValue()) request.AddParameter("source[cvc]", Cvc);
+                if (Name.HasValue()) request.AddParameter("source[name]", Name);
+                if (AddressLine1.HasValue()) request.AddParameter("source[address_line1]", AddressLine1);
+                if (AddressLine2.HasValue()) request.AddParameter("source[address_line2]", AddressLine2);
+                if (AddressCity.HasValue()) request.AddParameter("source[address_city]", AddressCity);
+                if (AddressState.HasValue()) request.AddParameter("source[address_state]", AddressState);
+                if (AddressZip.HasValue()) request.AddParameter("source[address_zip]", AddressZip);
+                if (AddressCountry.HasValue()) request.AddParameter("source[address_country]", AddressCountry);
+            }
+        }
+
+        public void AddParametersToRequest_Old(RestRequest request)
```

Comment:
I see that it is now called source rather than card for a charge in the docs. But a few questions:

1. Why keep around this method as "Old"? If the way to do things now is to use "source" instead of "card" I'd say we shouldn't keep this around. And it definitely doesn't make sense to make it public since the other version is explicitly implemented.
2. Do we need to increase the ApiVersion in StripeClient.cs for "source" to work?
3. Why is this change included in the PR for recipients?